controller/iteration.go

```diff
@@ -247,6 +247,12 @@ func (c *IterationController) Update(ctx *app.UpdateIterationContext) error {
 			if err != nil {
 				return jsonapi.JSONErrorResponse(ctx, err)
 			}
+			// New parent must not be one of existing children of subject iteration
+			for _, childItr := range oldSubtree {
+				if uuid.Equal(newParentIteration.ID, childItr.ID) {
```

Comment:
I read somewhere from the author of the uuid lib that you can just use the `==` comparator.

Reply:
okay, then let's start using `==` op henceforth 
changed in https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/4a007fb48f8f08955773a5da0a4638134e26b6b8